app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java

```diff
@@ -73,20 +76,39 @@ private void setAccessibilityServiceState() {
         }
     }
 
-    private ServerInstrumentation(Context context, int serverPort) {
+    private ServerInstrumentation(Context context, int serverPort, int mjpegServerPort) {
         if (!isValidPort(serverPort)) {
             throw new UiAutomator2Exception(String.format(
-                    "The port is out of valid range [%s;%s]: %s", MIN_PORT, MAX_PORT, serverPort));
+                "The server port is out of valid range [%s;%s]: %s",
+                MIN_PORT,
+                MAX_PORT,
+                serverPort
+            ));
         }
         this.serverPort = serverPort;
+
+        if (!isValidPort(mjpegServerPort)) {
+            Logger.warn(String.format(
+                "The mjpeg server port is out of valid range [%s;%s]: %s",
+                MIN_PORT,
+                MAX_PORT,
+                mjpegServerPort
+            ));
+        }
+        this.mjpegServerPort = mjpegServerPort;
```

Comment:
should we still set it if it is not in valid range?

Reply:
I thought about this when I was making the logging vs. exception change from the previous comments and think yes, because since both the `{start,stop}MjpegServer` method(s)  are now guarded w/ the `isValidPort` call as well.

Reply:
I mean maybe it makes sense to just keep the default value if the retrieved one is invalid?

Reply:
This will be resolved w/ the upcoming update(s)

Reply:
Sure, let's go w/ that. I'll update the logging accordingly.

Reply:
I did the same for `serverPort` as well for consistency -- please let me know what you think once I update.